services/src/main/java/org/keycloak/services/resources/admin/ProtocolMappersResource.java

```diff
@@ -49,8 +51,9 @@
     @Context
     protected KeycloakSession session;
 
-    public ProtocolMappersResource(ClientModel client, RealmAuth auth, AdminEventBuilder adminEvent) {
+    public ProtocolMappersResource(RealmModel realm, ClientModel client, RealmAuth auth, AdminEventBuilder adminEvent) {

```

Comment:
No need to pass in realm here to set in event. As I mentioned above just set it in RealmAdminResource and you're done